src/alert_manager.cpp

```diff
@@ -97,11 +96,8 @@ namespace libtorrent
 		}
 
 #ifndef TORRENT_DISABLE_EXTENSIONS
-		for (ses_extension_list_t::iterator i = m_ses_extensions.begin()
-			, end(m_ses_extensions.end()); i != end; ++i)
-		{
-			(*i)->on_alert(a);
-		}
+		notify_extensions(a, m_ses_extensions);
+		notify_extensions(a, m_ses_extensions_reliable);
```

Comment:
oh, I see. you call it twice. I would prefer to have ``m_ses_extensions`` be a superset and have `notify_extensions()` just called once. I can't think of a good argument right now though